### PR TITLE
Feature(CI): Add Windows Builds

### PR DESCRIPTION
**Pull request by magicaldave**
Not currently ready. Tossing the existing CI at github and seeing what it does.

### CODE REVIEW

**Comment by magicaldave** [comment]
OOOooooooooooohhhhhhhhhhhkaaaaaaaayyyyyy.

Many, many commits later, this is ready.

The architecture has been changed slightly for the build scripts. There are some problems with the CI itself, but from what I can tell I've assembled a complete package here, just in a sub-optimal way.

- The server does not yet build with IO2. We may take care of this via artifactory during scrum, alternatively another PR will just add it to the CMakeLists *and* the `make_package.msvc` script.
- Build_raknet.sh has been removed for linux builds altogether, but still exists for windows ones as the CMake for raknet doesn't currently 
- No unit tests, and I don't know how to download them currently
- CCache still doesn't work for windows. It's attempting to use the cache but all compilations are considered uncacheable atm. I attemped a patch in the main cmakelists based on the CCache documentation, but it seems not to have had any effect. Using sccache seems like it may also be a solution (SCCache stubbed in the workflow file)

Having said that, the CI is functional and complete with the exception of IO which I think is out of scope. One way or another, he's ready and a second optimization PR will need to be done to fix the above issues, because they take a very long time to test.

**Comment by magicaldave** [comment]
Ooooohhhhhkay. One last time on this one.

This now:

- Generates windows builds as artifacts
- uses ccache
- downloads prebuilt raknet
- downloads DreamWeave LuaJIT /w 5.2 extensions
- Caches its dependencies at build time

I'm still deeply unhappy that Windows builds require such a prodigious cache size for such little gain. This build uses almost 10X the storage of Linux jobs and still takes triple the time of a slow one. However, I don't believe it's going to get much better at this time, and I don't think further adjustments really need to be made until such time as we're simply not bothering with this build system.

I believe this should *now* be ready.

**Comment by magicaldave** [comment]
Okay, minor fib, I broke the unix builds.
Now it's ready.

File: .github/workflows/openmw-mp.yml [hunk 1]
```diff
@@ -1,19 +1,15 @@
-name: Ubuntu Build & Test
+name: CMake
```
Comment (HotaruBlaze) [inline .github/workflows/openmw-mp.yml:1]: May be a good idea to change this name, but thats a nitpick imo.

**Review by HotaruBlaze — APPROVED** [review]

File: .github/workflows/openmw-mp.yml (hunk 1 above)
Comment (magicaldave) [reply .github/workflows/openmw-mp.yml:1]: https://github.com/DreamWeave-MP/Dreamweave/blob/39e8e75ce902a7a4ca2958d6e127c510699e4be8/.github/workflows/openmw-mp.yml#L1

Comment (magicaldave) [reply .github/workflows/openmw-mp.yml:1]: Clearly there's some consistency issues with the last couple commits :P

**Review by HotaruBlaze — APPROVED** [review]

**Comment by HotaruBlaze** [comment]
I'm gonna approve it, but it'd probably be a good idea to squash this and see if it will break anything on master first.

**Comment by magicaldave** [comment]
Oh yeah. We're definitely not merging all these commits on top of master. It'll fuck up our history so bad if we do that, haha.

**Pull request merged**